python/dask_cudf/dask_cudf/tests/test_dispatch.py

```diff
@@ -38,3 +39,27 @@ def test_pyarrow_conversion_dispatch():
 
     assert type(df1) == type(df2)
     assert_eq(df1, df2)
+
+
+@pytest.mark.parametrize("index", [None, [1, 2] * 5])
+def test_deterministic_tokenize(index):
+    # Checks that `dask.base.normalize_token` correctly
+    # dispatches to the logic defined in `backends.py`
+    # (making `tokenize(<cudf-data>)` deterministic).
+    df = cudf.DataFrame(
+        {"A": list(range(10)), "B": ["dog", "cat"] * 5},
+        index=index,
+    )
+
+    # Matching data should produce the same token
+    assert tokenize(df) == tokenize(df)
+    assert tokenize(df.A) == tokenize(df.A)
+    assert tokenize(df.index) == tokenize(df.index)
+    assert tokenize(df) == tokenize(df.copy(deep=True))
+    assert tokenize(df.A) == tokenize(df.A.copy(deep=True))
+    assert tokenize(df.index) == tokenize(df.index.copy(deep=True))
+
+    # Modifying an element should change the token
+    original_token = tokenize(df.A)
+    df.A.iloc[2] = 10
+    assert original_token != tokenize(df.A)
```

Comment:
Can you test tokenizing an index as well as a dataframe?

Reply:
Ah, good call. It actually turns out that `hash_values` does **not** include the index in the hash, therefore we actually "need" `GenericIndex.__dask_tokenize__` (thought we can use caching for the index since it is immutable).